### ISSUE TITLE
reward_norm == 0 and no weights update for lora?

### ISSUE THREAD

**Issue by imyangs** [CaraJ7/CoMat#8]
Dear Author,

I tried to finetune sdxl with your code, but with the training progress continues the reward_norm becomes zero  which in my view means no grad will flow through the pipeline and no weights will update? (weights for the lora up layer is always zero)

I read some code but cannot figure out the reason. Can you give me some guidances? Thanks

<img width="428" alt="image" src="https://github.com/CaraJ7/CoMat/assets/28534716/53201025-4b24-4abb-89e6-5a4b8aad82e3">

**Comment by CaraJ7** [comment]
Hi @imyangs, thanks for reaching out. Just to make sure, do you follow the standard procedure for fine-tuning? You need to first fine-tune SDXL on 512 resolution, then generate the latents in Fidelity Preservation module, and finally follow all the settings of the hyperparameters in scripts/sdxl.sh?

**Comment by imyangs** [comment]
Thank you @CaraJ7  for your feedback. I skipped fine-tuning for SDXL at a 512 resolution to quick start. Through my rough debugging, I set gradient_accumulation_steps=1 to make the program run normally, but I believe I did not run the training correctly. I am looking forward to receiving more accurate guidance.

By the way, I fine-tuned SD1.5 with your code, but did not obtain good improvements. I want to know if there are any important considerations during training.
